File: aoc_elixir/day_01_pt2.js
let data = require("fs")
  .readFileSync("day_01_input.txt")
  .toString()
  .split(/\n/)

let solution = []
const numDefArr = [
  "zero",
  "one",
  "two",
  "three",
  "four",
  "five",
  "six",
  "seven",
  "eight",
  "nine",
]

for (let i = 0; i < data.length; i++) {
  let numArray = [
    ...data[i].matchAll(
      /(?=([0-9]|zero|one|two|three|four|five|six|seven|eight|nine))/g
    ),
  ]
  let parsed = numArray.map((x) => {
    return x[1]
  })
  //console.log(parsed)
  let intArray = parsed.map((x) => {
     if (!parseInt(x)) {
      return numDefArr.indexOf(x)
    } else return parseInt(x)
  })
  let numString = intArray.join("")

  let num = `${numString[0]}${numString.charAt(numString.length - 1)}`
  solution.push(parseInt(num))
  //console.log(numArray, numString, num)
}

function sum(acc, a) {
  return acc + a
}

console.log(solution.reduce(sum, 0))

File: aoc_elixir/day_01_pt2.exs
defmodule Solve do
    def solve(input) do
        #take the input and split by new lines
        String.split(input, "\n", trim: true)
        |> IO.inspect(label: "split values")
        #map over the input,and using RegEx match on anything that is a number or a eng string of a number
        |> Enum.map(fn str -> 
            Regex.scan(~r/[0-9]|zero|one|two|three|four|five|six|seven|eight|nine/, str)
            |> Enum.every(fn x -> x end)       
        end)
    def parse_input(file) do
        {:ok, output} = File.read(file)
        output
    end
    def spelled_to_int(str) do
        int_key = %{
            "zero" => 0,
            "one" => 1,
            "two" => 2,
            "three" => 3,
            "four" => 4,
            "five" => 5,
            "six" => 6,
            "seven" => 7,
            "eight" => 8,
            "nine" => 9
        }
        int_key[str]
    end
end

input = """
two1nine
eightwothree
abcone2threexyz
xtwone3four
4nineeightseven2
zoneight234
7pqrstsixteen
"""

IO.puts(Solve.solve(input))
